src/networkd.c

```diff
@@ -1398,6 +1452,48 @@ _netplan_netdef_write_networkd(
     if (!_netplan_netdef_write_network_file(np_state, def, rootdir, path_base, has_been_written, error))
         return FALSE;
     SET_OPT_OUT_PTR(has_been_written, TRUE);
+
+    /* When activation-mode is used we default to being optional.
+     * Otherwise, systemd might wait indefinitely for the interface to
+     * come online.
+     */
+    if (!(def->optional || def->activation_mode)) {
```

Comment:
question: Should we also check if IP `addresses` or `dhcp4`/`dhcp6` are defined, e.g. to ignore definitions like this `ethernets.encc000: {}`?

We could define such scenarios as "user error", or try to be smart about it..

Reply:
I'd probably opt to "user error" for now (the easy workaround is adding `optional: true` instead of `{}`) and in the future we might want to wait for different conditions, e.g. link-layer configuration, which would still involve  some waiting, but not necessarily on routable IPs.

Reply:
I'm not sure if I understood the question. Are you suggesting we shouldn't create the link for that type of definition (empty)?

Reply:
Empty NetDefs will not reach the "online" state, due to lacking an IP address. I was wondering if we should implicitly mark them "optional: true". But I now feel like we shouldn't do it, as logic might change in the future with the implementation of "optional-addresses".